rmw_fastrtps_cpp/src/rmw_publisher.cpp

```diff
@@ -128,7 +128,8 @@ rmw_create_publisher(
     goto fail;
   }
 
-  info->publisher_ = Domain::createPublisher(participant, publisherParam, nullptr);
+  info->listener_ = new PubListener(info);
```

Comment:
The use of `new` here is actually dangerous, because it will throw if malloc fails, so it will try to throw an exception across the rmw C API, which is undefined behavior I believe.

Reply:
See this for how new and nothrow work together: https://en.cppreference.com/w/cpp/memory/new/nothrow